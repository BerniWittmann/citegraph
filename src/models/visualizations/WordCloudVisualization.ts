import Visualization, { VisualizationBaseFields } from './Visualization'

export interface WordCloudVisualizationFields {}

export default class WordCloudVisualization extends Visualization implements WordCloudVisualizationFields {
  static key: string = 'wordcloud'
  key: string = 'wordcloud'
  static title: string = 'visualizations.information.word_cloud.title'
  static description: string = 'visualizations.information.word_cloud.description'
  static longDescription: string = 'visualizations.information.word_cloud.long_description'
  static imageUrl: string = 'visualization_images/word_cloud.jpg'

  constructor ({ id, name, progress }: VisualizationBaseFields & WordCloudVisualizationFields) {
    super({ id, name, progress })
  }
}
