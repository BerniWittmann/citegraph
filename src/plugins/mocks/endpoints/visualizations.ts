import { AxiosRequestConfig } from 'axios'
import MockAdapter from 'axios-mock-adapter'
import db from '../db'
import Project from '@/models/project'

const projects = db.getCollection('projects')
const visualizations = db.getCollection('visualizations')

function getProjectId (config: AxiosRequestConfig): number {
  const urlParts: Array<string> = config.url!.split('/')
  return parseInt(urlParts[4])
}

export function setupMocks (mock: MockAdapter) {
  mock.onGet(/\/projects\/\d+\/visualizations$/).reply((config: AxiosRequestConfig) => {
    const project: Resultset<Project> = projects.findOne({ 'id': getProjectId(config) })
    if (!project) {
      return [404]
    }
    return [200, visualizations.find()]
  })
}
