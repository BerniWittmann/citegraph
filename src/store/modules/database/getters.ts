import { GetterTree } from 'vuex'
import { RootState } from '@/store/types'
import { DatabaseState } from '@/store/modules/database/types'

export const getters: GetterTree<DatabaseState, RootState> = {}
