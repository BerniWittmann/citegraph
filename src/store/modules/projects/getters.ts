import { GetterTree } from 'vuex'
import { ProjectsState } from '@/store/modules/projects/types'
import { RootState } from '@/store/types'
import Project from '@/models/project'

export const getters: GetterTree<ProjectsState, RootState> = {
  activeProject (state: ProjectsState): Project | undefined {
    return state.activeProject
  },
  hasActiveProject (state: ProjectsState): boolean {
    return !!state.activeProject
  },
  projects (state: ProjectsState): Array<Project> {
    return state.projects
  },
  openProjects (state: ProjectsState): Array<Project> {
    return state.openProjects
  }
}
