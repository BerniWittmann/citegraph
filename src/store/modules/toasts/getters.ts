import { GetterTree } from 'vuex'
import { RootState } from '@/store/types'
import { ToastsState } from '@/store/modules/toasts/types'
import Toast from '@/models/toast'

export const getters: GetterTree<ToastsState, RootState> = {
  toasts (state: ToastsState): Array<Toast> {
    return state.toasts
  }
}
