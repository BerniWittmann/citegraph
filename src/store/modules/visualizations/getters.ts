import { GetterTree } from 'vuex'
import { VisualizationsState } from './types'
import { RootState } from '@/store/types'
import Visualization from '@/models/visualizations/Visualization'

export const getters: GetterTree<VisualizationsState, RootState> = {
  visualizations (state: VisualizationsState): Array<Visualization> {
    return state.visualizations
  }
}
