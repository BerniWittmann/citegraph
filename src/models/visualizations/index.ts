import WordCloudVisualization, { WordCloudVisualizationFields } from './WordCloudVisualization'
import { VisualizationBaseFields } from '@/models/visualizations/Visualization'
import BarChartVisualization, { BarChartVisualizationFields } from '@/models/visualizations/BarChartVisualization'

export const visualizations = [WordCloudVisualization, BarChartVisualization]

const visualizationsKeyMap: Record<string, any> = {}
visualizations.forEach((visualization) => {
  visualizationsKeyMap[visualization.key] = visualization
})

export {
  visualizationsKeyMap
}

export type VisalizationFields = VisualizationBaseFields & (
  WordCloudVisualizationFields &
  BarChartVisualizationFields
)
