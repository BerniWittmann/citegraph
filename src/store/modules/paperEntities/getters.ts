import { GetterTree } from 'vuex'
import { RootState } from '@/store/types'
import { PaperEntitiesState } from './types'
import { PaperEntity } from '@/models/paperEntities'

export const getters: GetterTree<PaperEntitiesState, RootState> = {
  entities (state: PaperEntitiesState): Array<PaperEntity> {
    return state.entities
  },

  entityType (state: PaperEntitiesState): string | undefined {
    return state.entityType
  },

  entityCount (state: PaperEntitiesState): number {
    return state.entityCount
  },

  activeEntity (state: PaperEntitiesState): PaperEntity | undefined {
    return state.activeEntity
  },

  hasActiveEntity (state: PaperEntitiesState): boolean {
    return !!state.activeEntity
  }
}
