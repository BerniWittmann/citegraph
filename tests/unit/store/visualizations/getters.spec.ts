import { getters } from '@/store/modules/visualizations/getters'
import WordCloudVisualization from '@/models/visualizations/WordCloudVisualization'

describe('store/modules/visualizations/getters', () => {
  it('provides the getters', () => {
    expect(getters).toMatchSnapshot()
  })

  describe('visualizations', () => {
    it('returns the stored visualizations', () => {
      const state = {
        visualizations: [
          new WordCloudVisualization({ id: '1', name: 'First Project' }),
          new WordCloudVisualization({ id: '2', name: 'Second Project' })
        ]
      }
      const getter = getters.visualizations as Function

      expect(getter(state)).toEqual(state.visualizations)
    })

    it('returns empty array if no visualizations are available', () => {
      const state = {
        visualizations: []
      }
      const getter = getters.visualizations as Function

      expect(getter(state)).toEqual([])
    })
  })
})
