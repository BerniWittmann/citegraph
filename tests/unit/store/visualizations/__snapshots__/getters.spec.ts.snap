// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store/modules/visualizations/getters provides the getters 1`] = `
Object {
  "visualizations": [Function],
}
`;
